lib-injection/sources/sitecustomize.py

```diff
@@ -282,17 +296,18 @@ def _inject():
                     "DD_INJECT_FORCE set to True, allowing unsupported runtimes and continuing.",
                     level="debug",
                 )
-        if telemetry_data:
+        if runtime_incomp or denylist_package:
```

Comment:
nit: Could we just do this instead? And remove the `denylist_package` flag?
```suggestion
        if runtime_incomp or not FORCE_INJECT:
```